### PR TITLE
[ODPR-634] Updated support for registration with ID.

### PR DESCRIPTION
**Pull request by erasmos-hmrc-new**

### CODE REVIEW

File: app/uk/gov/hmrc/dprs/services/registration/withId/RegistrationWithIdService.scala [hunk 1]
```diff
@@ -81,6 +84,14 @@ object RegistrationWithIdService {
...
           (JsPath \ "emailAddress").writeNullable[String])(unlift(ContactDetails.unapply))
     }
+
+    object ConnectorErrorCode {
+      val badRequest          = "400"
+      val duplicateSubmission = "409"
+      val noMatch             = "404"
+      val couldNotBeProcessed = "503"
+      val internalError       = "500"
+    }
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc-new) [reply app/uk/gov/hmrc/dprs/services/registration/withId/RegistrationWithIdService.scala:94]: These are the error codes from the EIS APIs, which you'll find in the stubs and specs; they are not controlled by us; and note that the meaning of the codes sometimes change according to the API (see `couldNotBeProcessed` in `RegistrationWithIdService` vs `CreateSubscriptionService`) - which is why I had to revert from that abstraction, I introduced recently.

File: app/uk/gov/hmrc/dprs/services/registration/withoutId/RegistrationWithoutIdService.scala [hunk 2]
```diff
@@ -119,5 +118,11 @@ object RegistrationWithoutIdService {
       implicit val writes: OWrites[Error] =
         (JsPath \ "code").write[String].contramap(_.code)
     }
+
+    object ConnectorErrorCode {
+      val internalServerError = "500"
+      val serviceUnavailable  = "503"
+    }
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc-new) [reply app/uk/gov/hmrc/dprs/services/registration/withoutId/RegistrationWithoutIdService.scala:125]: See above.

File: app/uk/gov/hmrc/dprs/services/subscription/CreateSubscriptionService.scala [hunk 3]
```diff
@@ -175,6 +175,16 @@ object CreateSubscriptionService {
...
         (JsPath \ "id").write[String].contramap(_.id)
+
+      object ConnectorErrorCode {
+        val couldNotBeProcessed = "003"
+        val duplicateSubmission = "004"
+        val forbidden           = "403"
+        val invalidId           = "016"
+        val malformedPayload    = "400"
+        val unauthorised        = "401"
+      }
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc-new) [reply app/uk/gov/hmrc/dprs/services/subscription/CreateSubscriptionService.scala:186]: See above.

[the remark this answers is not in the document]
Comment (erasmos-hmrc-new) [reply app/uk/gov/hmrc/dprs/services/subscription/CreateSubscriptionService.scala:186]: See above.

File: app/uk/gov/hmrc/dprs/services/subscription/ReadSubscriptionService.scala [hunk 4]
```diff
@@ -116,6 +114,17 @@ object ReadSubscriptionService {
...
+      object ConnectorErrorCode {
+        val couldNotBeProcessed     = "003"
+        val createOrAmendInProgress = "201"
+        val duplicateSubmission     = "004"
+        val forbidden               = "403"
+        val internalServerError     = "500"
+        val invalidId               = "016"
+        val noSubscription          = "202"
+        val unauthorised            = "401"
+      }
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc-new) [reply app/uk/gov/hmrc/dprs/services/subscription/ReadSubscriptionService.scala:127]: See above.

File: app/uk/gov/hmrc/dprs/services/subscription/UpdateSubscriptionService.scala [hunk 5]
```diff
@@ -133,6 +135,18 @@ object UpdateSubscriptionService {
...
+    object Response {
+      object ConnectorErrorCode {
+        val couldNotBeProcessed     = "003"
+        val createOrAmendInProgress = "201"
+        val duplicateSubmission     = "004"
+        val forbidden               = "403"
+        val internalServerError     = "500"
+        val noSubscription          = "202"
+        val unauthorised            = "401"
+      }
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc-new) [reply app/uk/gov/hmrc/dprs/services/subscription/UpdateSubscriptionService.scala:148]: See above.

**Review by paddynski-opencast — APPROVED** [review]

**Pull request merged**